x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -80,16 +87,31 @@
 public interface AuthorizationEngine {
 
     /**
-     * Asynchronously resolves any necessary information to authorize the given user(s). This could
-     * include retrieval of permissions from an index or external system.
+     * Asynchronously resolves the information necessary to authorize the given request, which has
+     * already been authenticated. This could include retrieval of permissions from an index or external system.
+     * See also {@link #resolveAuthorizationInfo(Subject, ActionListener)}, for which this method is the more
+     * specific sibling. This returns the specific {@code AuthorizationInfo} used to authorize only the specified request.
      *
-     * @param requestInfo object contain the request and associated information such as the action
+     * @param requestInfo object containing the request and associated information such as the action name
      *                    and associated user(s)
      * @param listener the listener to be notified of success using {@link ActionListener#onResponse(Object)}
      *                 or failure using {@link ActionListener#onFailure(Exception)}
      */
     void resolveAuthorizationInfo(RequestInfo requestInfo, ActionListener<AuthorizationInfo> listener);
 
+    /**
+     * Asynchronously resolves the information necessary to authorize requests in the context of the given {@code Subject}.
+     * This could include retrieval of permissions from an index or external system.
+     * See also {@link #resolveAuthorizationInfo(RequestInfo, ActionListener)}, for which this method is the more general
+     * sibling. This returns the {@code AuthorizationInfo} that is used for access checks outside the context of
+     * authorizing a specific request, i.e. {@link #checkPrivileges(AuthorizationInfo, PrivilegesToCheck, Collection, ActionListener)}
+     *
+     * @param subject object representing the effective user
+     * @param listener the listener to be notified of success using {@link ActionListener#onResponse(Object)}
+     *                 or failure using {@link ActionListener#onFailure(Exception)}
+     */
+    void resolveAuthorizationInfo(Subject subject, ActionListener<AuthorizationInfo> listener);
```

Comment:
[CustomAuthorizationEngine](https://github.com/elastic/elasticsearch/blob/master/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java) needs to be updated as well for these interface changes. Otherwise CI will get compilation failures once this PR is merged (similar to https://github.com/elastic/elasticsearch/issues/86378). It does not fail here because it [does not run](https://github.com/elastic/elasticsearch/issues/86378#issuecomment-1116630079) on regular PR.

Reply:
Nice catch, pushed [11d0c88](https://github.com/elastic/elasticsearch/pull/85898/commits/11d0c889953d17e376ca3a89e2b2109832f39b99) , but I didn't manage to verify that it actually works 🤞